### PR TITLE
Issue #83 edit remove

### PR DESCRIPTION
**Pull request by Agri**

### CODE REVIEW

File: src/projects-screen/controls/project-definition-modal/index.tsx [hunk 1]
```diff
@@ -43,36 +45,100 @@ class ProjectDefinitionModalComponent extends React.Component<IAddProjectModalCo
...
-      const channel: IChannel = _.isArray(nextProps.channels) && !_.isEmpty(nextProps.channels) ?
-        _.head(nextProps.channels) : { id: '0', name: '' } as IChannel;
+      let projectName;
+      let siteId;
+      let siteName;
+      let finalChannelId;
+      let finalChannelName;
+      let rawChannelId;
+      let rawChannelName;
+
```
[the remark this answers is not in the document]
Comment (Agri) [reply src/projects-screen/controls/project-definition-modal/index.tsx]: ok, right, forgot to clean it

File: src/projects-screen/controls/project-definition-modal/index.tsx [hunk 2]
```diff
@@ -43,36 +45,73 @@ class ProjectDefinitionModalComponent extends React.Component<IAddProjectModalCo
...
+        siteName,
+        finalChannelId,
+        finalChannelName,
+        rawChannelId,
+        rawChannelName,
       });
     } else if (this.modalIsShown(nextProps)) {
-      const channel: IChannel = _.isArray(nextProps.channels) && !_.isEmpty(nextProps.channels) ?
-        _.head(nextProps.channels) : { id: '0', name: '' } as IChannel;
+      let finalChannelId;
```
Comment (sharpcoding) [inline src/projects-screen/controls/project-definition-modal/index.tsx]: so how about changing this one as well ?

**Comment by Agri** [comment]
moved some code in place where it's used

File: src/projects-screen/controls/project-definition-modal/index.tsx [hunk 3]
```diff
@@ -71,6 +72,7 @@ class ProjectDefinitionModalComponent extends React.Component<IAddProjectModalCo
         finalChannelName,
         rawChannelId,
         rawChannelName,
+        siteChanged: false,
```
Comment (sharpcoding) [inline src/projects-screen/controls/project-definition-modal/index.tsx]: This is not necessary

Btw. I don't find else if's a clean code so suggest to change them

```ts
private sitesDropdownReloads = (nextProps: IAddProjectModalComponentProps): boolean =>
  JSON.stringify(this.props.sites) !== JSON.stringify(nextProps.sites);

///....

public componentWillReceiveProps(nextProps: IAddProjectModalComponentProps & IAddProjectModalComponentActionCreators) {
  if (this.modalWillAppear(nextProps)) {
    const id = _.isEmpty(nextProps.editedProject.id) ? '' : nextProps.editedProject.id;
    const projectName = _.isEmpty(nextProps.editedProject.projectName) ? '' : nextProps.editedProject.projectName;
    const siteId = _.isEmpty(nextProps.editedProject.siteId) ? '' : nextProps.editedProject.siteId;
    const siteName = _.isEmpty(nextProps.editedProject.siteName) ? '' : nextProps.editedProject.siteName;
    const finalChannelId = _.isEmpty(nextProps.editedProject.finalChannelId) ? '' : nextProps.editedProject.finalChannelId;
    const finalChannelName = _.isEmpty(nextProps.editedProject.finalChannelName) ? '' : nextProps.editedProject.finalChannelName;
    const rawChannelId = _.isEmpty(nextProps.editedProject.rawChannelId) ? '' : nextProps.editedProject.rawChannelId;
    const rawChannelName = _.isEmpty(nextProps.editedProject.rawChannelName) ? '' : nextProps.editedProject.rawChannelName;

    this.setState({
      id,
      projectName,
      siteId,
      siteName,
      finalChannelId,
      finalChannelName,
      rawChannelId,
      rawChannelName,
    });
    return;
  }
  if (this.modalIsShown(nextProps)) {
    if (this.sitesDropdownReloads(nextProps)) {
      return;
    }

    const cannotFindChannelInProps = (channelId) => !_.includes(_.map(this.props.channels, (ch) => ch.id), channelId) &&
                                                    !_.includes(_.map(nextProps.channels, (ch) => ch.id), channelId);

    const firstChannel: IChannel = _.isArray(nextProps.channels) && !_.isEmpty(nextProps.channels) ?
      _.head(nextProps.channels) : { id: '0', name: '' } as IChannel;

    const finalChannelId = cannotFindChannelInProps(this.state.finalChannelId) ? firstChannel.id : nextProps.editedProject.finalChannelId;
    const finalChannelName = cannotFindChannelInProps(this.state.finalChannelId) ? firstChannel.name : nextProps.editedProject.finalChannelName;
    const rawChannelId = cannotFindChannelInProps(this.state.rawChannelId) ? firstChannel.id : nextProps.editedProject.rawChannelId;
    const rawChannelName = cannotFindChannelInProps(this.state.rawChannelId) ? firstChannel.name : nextProps.editedProject.rawChannelName;

    this.setState({
      finalChannelId,
      finalChannelName,
      rawChannelId,
      rawChannelName,
    });
  }
}

```

**Pull request merged**